codec-http3/src/main/java/io/netty/handler/codec/http3/Http3Settings.java

```diff
@@ -0,0 +1,374 @@
+     * Controls the maximum size of the dynamic table used by QPACK.
+     */
+    public static final long HTTP3_SETTINGS_QPACK_MAX_TABLE_CAPACITY = 0x1;
+
+    /**
+     * Maximum field section size setting identifier (<b>0x6</b>).
+     * <p>
+     * Defined in <a href="https://datatracker.ietf.org/doc/html/rfc9114#section-7.2.4.1">
+     * RFC 9114, Section 7.2.4.1 (SETTINGS_MAX_FIELD_SECTION_SIZE)</a> , also referenced
+     * in the <a href="https://datatracker.ietf.org/doc/html/rfc9114#section-7.2.4.1">
+     * HTTP/3 SETTINGS registry (RFC 9114, Section 7.2.4.1)</a> and registered in
+     * the <a href="https://www.iana.org/assignments/http3-parameters/http3-parameters.xhtml#settings">
+     * HTTP/3 SETTINGS registry (IANA)</a>.
+     * <br>
+     * Specifies the upper bound on the total size of HTTP field sections accepted by a peer.
+     */
+    public static final long HTTP3_SETTINGS_MAX_FIELD_SECTION_SIZE = 0x6;
+
+    /**
+     * QPACK blocked streams setting identifier (<b>0x7</b>).
+     * <p>
+     * Defined in <a href="https://datatracker.ietf.org/doc/html/rfc9204#section-5">
+     * RFC 9204, Section 5 (QPACK_BLOCKED_STREAMS)</a> and registered in
+     * the <a href="https://www.iana.org/assignments/http3-parameters/http3-parameters.xhtml#settings">
+     * HTTP/3 SETTINGS registry (IANA)</a>.
+     * <br>
+     * Indicates the maximum number of streams that can be blocked waiting for QPACK instructions.
+     */
+    public static final long HTTP3_SETTINGS_QPACK_BLOCKED_STREAMS = 0x7;
+
+    /**
+     * ENABLE_CONNECT_PROTOCOL setting identifier (<b>0x8</b>).
+     * <p>
+     * Defined and registered in <a href="https://datatracker.ietf.org/doc/html/rfc9220#section-5">
+     * RFC 9220, Section 5 (IANA Considerations)</a> and registered in
+     * the <a href="https://www.iana.org/assignments/http3-parameters/http3-parameters.xhtml#settings">
+     * HTTP/3 SETTINGS registry (IANA)</a>.
+     * <br>
+     * Enables use of the CONNECT protocol in HTTP/3 when set to 1; disabled when 0.
+     */
+    public static final long HTTP3_SETTINGS_ENABLE_CONNECT_PROTOCOL = 0x8;
+
+    private static final Long TRUE = 1L;
+    private static final Long FALSE = 0L;
+
+    /**
+     * Creates a new instance with an initial capacity of 4 entries.
+     */
+    public Http3Settings() {
+        this.settings = new LongObjectHashMap<>(4);
+    }
+
+    /**
+     * Creates a new instance with the specified initial capacity.
+     *
+     * @param initialCapacity initial capacity of the underlying map
+     */
+    public Http3Settings(int initialCapacity) {
+        this.settings = new LongObjectHashMap<>(initialCapacity);
+    }
+
+    /**
+     * Creates a new instance with the specified initial capacity and load factor.
+     *
+     * @param initialCapacity initial capacity of the underlying map
+     * @param loadFactor load factor for the underlying map
+     */
+    public Http3Settings(int initialCapacity, float loadFactor) {
```

Comment:
I would suggest to start with package-private for now... if people really need this constructor we can add it later on.
```suggestion
   Http3Settings(int initialCapacity, float loadFactor) {
```

Reply:
made as package private